### PR TITLE
Video-16-Build-Cart-Screen

### PR DESCRIPTION
**Pull request by basir**

### CODE REVIEW

File: frontend/src/screens/CartScreen.js [hunk 1]
```diff
@@ -1,24 +1,106 @@
...
+    <div className="row top">
+      <div className="col-2">
+        <h1>Shopping Cart</h1>
+        {cartItems.length === 0 ? (
+          <MessageBox>
+            Cart is empty. <Link to="/">Go Shopping</Link>
+          </MessageBox>
+        ) : (
+          <ul>
+            {cartItems.map((item) => (
```
Comment (ghost) [inline frontend/src/screens/CartScreen.js:38]: facing a issue in this line-
Uncaught TypeError: cartItems.map is not a function
can anyone help? @basir

**Review by yogesh9987 — COMMENTED** [review]
cartItems: state.cartItems.map((x) =>
            x.product === existItem.product ? item : x
          ),
error

File: frontend/src/screens/CartScreen.js (hunk 1 above)
Comment (FarahatDeploys) [reply frontend/src/screens/CartScreen.js:38]: this means that there is issue with cartItems you should log console.log(cartItems) and see what is stored there

**Review by yogesh9987 — COMMENTED** [review]
No routes matched location "/cart/123?qty=4"

**Review by yogesh9987 — COMMENTED** [review]
cartItems.map is not funcation

File: frontend/src/screens/CartScreen.js (hunk 1 above)
Comment (basir) [reply frontend/src/screens/CartScreen.js:38]: follow @FarahatHadi guide.

**Comment by Dewruwan95** [comment]
here it is not visible my product quantity but in total item count, it is visible total product quantity. all select dropdowns are showing only 1. I try to find errors by comparing my code and this code. but I couldn't find any mistake. please help me to fix this issue.
image

**Pull request merged**